pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -97,6 +98,10 @@ func (p *Probe) checkUpdated() error {
 
 // checkServerInfo checks to ensure that Envoy is in the READY state
 func (p *Probe) checkServerInfo() error {
+	// If Envoy is already Live, just return immediately.
+	if p.serverLive {
```

Comment:
It seems once live, it is always live. But we have cases envoy hot restart.

So if this api call is really costly, we can cache it for a period.

Reply:
Envoy hot restart does not change live ness of that Envoy process.

Reply:
Understood